src/api/controllers/reportedCommentAdminController.js

```diff
@@ -0,0 +1,107 @@
+import models from '../models';
+import status from '../../helpers/constants/status.codes';
+import paginateUser from '../../helpers/generate.pagination.details';
+
+const generatePagination = paginateUser;
+
+const {
+  Comment, ReportedComment, User, Reason
+} = models;
+
+/**
+ * contains all method for an admin to read/delete/Block/unblock a comment
+ *
+ * @export
+ * @class AdminReportedCommentController
+ */
+class AdminReportedCommentController {
+  /**
+   * This function gets all the users that are stored in the User table
+   *
+   * @author Frank Mutabazi
+   * @static
+   * @param {Object} req the request
+   * @param {Object} res the response to be sent
+   * @memberof AdminReportedCommentController
+   * @returns {Object} res
+   */
+  static async getAllReportedComments(req, res) {
+    const defaultOffset = 0;
+    const defaultLimit = 10;
+    const offset = req.query.offset || defaultOffset;
+    const limit = req.query.limit || defaultLimit;
+    const result = await ReportedComment.findAll({
+      attributes: ['createdAt'],
+      include: [
+        {
+          model: Comment,
+          required: true,
+          attributes: ['body', 'articleSlug']
+        },
+        {
+          model: User,
+          required: true,
+          attributes: ['username', 'email']
+        },
+        {
+          model: Reason,
+          required: true,
+          attributes: ['description']
+        }
+      ],
+      offset,
+      limit
+    });
+    return res.status(status.OK).send({
+      status: status.OK,
+      paginationDetails: generatePagination(result.length,
+        result,
+        offset,
+        limit),
+      data: result
+    });
+  }
+
+  /**
+   * fetch a single reported Comment
+   *
+   * @author Frank Mutabazi
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - respond object
+   * @returns {array} response body
+   * @memberof AdminReportedCommentController
+   */
+  static async singleReportedComment(req, res) {
+    const { slug } = req.params;
+    const response = await ReportedComment.findOne({
+      attributes: ['createdAt'],
+      where: {
+        reportedArticleSlug: slug
```

Comment:
are you sure this should be using the slug to find a particular comment? because according to what you did, this will be fetching a comment for a particular article. which will be returning the same comment all the time